### ISSUE TITLE
What was numpy 2.0.0?

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#9066]
Theres [a milestone](https://github.com/numpy/numpy/milestone/9) referencing this, along with various comments in the scipy source.

Was this a major revision that was decided to be a bad idea, or simply a quirk in the version numbering? If the latter, what version did 2.0.0 end up actually being?

**Comment by pv** [comment]
I think "Numpy 2.0" just informally means a hypothetical, backward-compatibility breaking redesign of Numpy somewhere in the far future.

**Comment by seberg** [comment]
Yeah, though my guess is, the "NumPy 2.0" idea has not really been anything more then a possible idea since at least numpy 1.7. So, basically the whole thing is just an archaeological detail at this time.

**Comment by charris** [comment]
There was discussion of an API breaking refactor of NumPy for some time back in the early days, and there have been three attempts at what might be labeled NumPy 2.0 that I know of. The first was Enthought, funded by Microsoft, back around 2010-2011, the numpy-refactor repository here is a fork of the Enthought work. @mwiebe  had a plan to move NumPy forward, involving moving to C++, expanding dtypes, and much else, and that foundered after the missing values kerfuffle. That kerfuffle pretty much put an end to NumPy making major compatibility breaks.  Mark continued developing his ideas on the side during his stint at Continuum,  Blaze is the result.

In case you are curious, the missing values idea was essentially to move masking into the ndarray base class. The ufunc `where` keyword is one of several artifacts of that.

That is the short history as I recall it. @teoliphant could probably fill in some details and pin down the dates. At this point, NumPy 2.0 is a creature of mythology.

**Comment by shoyer** [comment]
> Mark continued developing his ideas on the side during his stint at Continuum, Blaze is the result.

While at one point this was called "Blaze", Mark's work lives on in as [DyND](https://github.com/libdynd/dynd-python). The Python [Blaze](https://github.com/blaze/blaze) project no longer uses DyND.

**Comment by eric-wieser** [comment]
Some more info is here: https://web.archive.org/web/20130402032018/http://projects.scipy.org/numpy/roadmap

**Comment by a90091343** [comment]
After 6 years, numpy 2.0.0 has finally arrived. Congratulations!
